typescript/cli/src/commands/warp.ts

```diff
@@ -446,38 +446,50 @@ export const rebalancer: CommandModuleWithContext<{
     checkFrequency,
     strategyConfigFile,
   }) => {
-    try {
-      // Instantiates the warp route monitor
-      const monitor: IMonitor = new Monitor(
-        context.registry,
-        warpRouteId,
-        checkFrequency,
-      );
+    // Instantiates the warp route monitor
+    const monitor: IMonitor = new Monitor(
+      context.registry,
+      warpRouteId,
+      checkFrequency,
+    );
 
+    try {
       // Instantiates the strategy that will get rebalancing routes based on monitor results
       const strategy: IStrategy = Strategy.fromConfigFile(strategyConfigFile);
 
       // Instantiates the executor that will process rebalancing routes
       const executor: IExecutor = new Executor();
 
-      // Observe monitor events and process rebalancing routes
-      monitor.subscribe((event) => {
-        const balances = event.balances.reduce((acc, next) => {
-          acc[next.chain] = next.value;
-          return acc;
-        }, {} as Record<ChainName, bigint>);
-
-        const rebalancingRoutes = strategy.getRebalancingRoutes(balances);
-
-        executor.processRebalancingRoutes(rebalancingRoutes);
-      });
-
-      // Starts the monitor to begin polling balances.
-      await monitor.start();
-
-      logGreen('Rebalancer started successfully 🚀');
+      await monitor
+        // Observe balances events and process rebalancing routes
+        .on('collateralbalances', (event) => {
+          const balances = event.balances.reduce((acc, next) => {
+            acc[next.chain] = next.value;
+            return acc;
+          }, {} as Record<ChainName, bigint>);
+
+          const rebalancingRoutes = strategy.getRebalancingRoutes(balances);
+
+          executor.processRebalancingRoutes(rebalancingRoutes).catch((e) => {
+            throw new Error(
+              `Error processing rebalancing routes: ${e.messages}`,
+            );
+          });
+        })
+        // Observe monitor errors and exit
+        .on('error', (e) => {
+          throw new Error(
```

Comment:
Can we just log the error? Maybe the monitor just fails to poll some data, doesn't need to exit the process as it might recover on the next poll

Reply:
but this changes the original design of bubling the errors to this catch.

Maybe we can structure the errors and depending on the type log or throw.

That, I'll do it in a different PR/issue.